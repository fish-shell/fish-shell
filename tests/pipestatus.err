
####################
# pipestatus variable - builtins only

####################
# pipestatus variable - no builtins

####################
# pipestatus variable - mixed

####################
# pipestatus variable - non-pipe
