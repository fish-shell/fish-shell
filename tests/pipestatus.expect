# vim: set filetype=expect:
#
# Verify `$pipestatus` is preserved to next prompt.

spawn $fish

expect_prompt

# no pipe
send_line "true"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "0 : 0" {}
send_line "false"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "1 : 1" {}

# no pipe but negated
send_line "! true"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "0 : 1" {}
send_line "! false"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "1 : 0" {}

# pipe
send_line "false | true | false"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "1 0 1 : 1" {}

# pipe negated
send_line "! false | true | false"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "1 0 1 : 0" {}

# pipe mixed with builtins and external
send_line "command true | false | true | true"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "0 1 0 0 : 0" {}
send_line "command true | command false | command false"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "0 1 1 : 1" {}
send_line "sh -c 'exit 3' | command false | sh -c 'exit 5'"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "3 1 5 : 5" {}

# negated pipe with mixed builtin and external
send_line "! sh -c 'exit 3' | command false | sh -c 'exit 5'"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "3 1 5 : 0" {}

# block
send_line "command false | begin; command true; end | true"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "1 0 0 : 0" {}
send_line "command false | begin; ! true; end | true"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "1 1 0 : 0" {}

# syntax error
send_line "syntax terror"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "127 : 127" {}
send_line "syntax | terror"
expect_prompt
send_line "echo \$pipestatus : \$status"
expect_prompt "127 : 127" {}
