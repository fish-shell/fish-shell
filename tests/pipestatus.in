logmsg "pipestatus variable - builtins only"

false | false | false; echo $pipestatus
false | false | false; echo $status

true | true | true; echo $pipestatus
true | true | true; echo $status

false | true | false; echo $pipestatus
false | true | false; echo $status

true | false | true; echo $pipestatus
true | false | true; echo $status

logmsg "pipestatus variable - no builtins"

command false | command false | command false; echo $pipestatus
command false | command false | command false; echo $status

command true | command true | command true; echo $pipestatus
command true | command true | command true; echo $status

command false | command true | command false; echo $pipestatus
command false | command true | command false; echo $status

command true | command false | command true; echo $pipestatus
command true | command false | command true; echo $status

logmsg "pipestatus variable - mixed"

command false | command false | false; echo $pipestatus
command false | command false |  false; echo $status

command true | true | command true; echo $pipestatus
command true | true | command true; echo $status

false | command true | command false; echo $pipestatus
false | command true | command false; echo $status

true | false | command true; echo $pipestatus
true | false | command true; echo $status

sh -c 'exit 5' | sh -c 'exit 2'; echo $pipestatus
sh -c 'exit 5' | sh -c 'exit 2'; echo $status

sh -c 'exit 3' | false | sh -c 'exit 6'; echo $pipestatus
sh -c 'exit 3' | false | sh -c 'exit 6'; echo $status

sh -c 'exit 9' | true | sh -c 'exit 3' | false; echo $pipestatus
sh -c 'exit 9' | true | sh -c 'exit 3' | false; echo $status

logmsg "pipestatus variable - non-pipe"

true; echo $pipestatus
true; echo $status

false; echo $pipestatus
false; echo $status

command true; echo $pipestatus
command true; echo $status

command false; echo $pipestatus
command false; echo $status

sh -c 'exit 4'; echo $pipestatus
sh -c 'exit 4'; echo $status
