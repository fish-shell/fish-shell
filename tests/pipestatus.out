
####################
# pipestatus variable - builtins only
1 1 1
1
0 0 0
0
1 0 1
1
0 1 0
0

####################
# pipestatus variable - no builtins
1 1 1
1
0 0 0
0
1 0 1
1
0 1 0
0

####################
# pipestatus variable - mixed
1 1 1
1
0 0 0
0
1 0 1
1
0 1 0
0
5 2
2
3 1 6
6
9 0 3 1
1

####################
# pipestatus variable - non-pipe
0
0
1
1
0
0
1
1
4
4
